iwyu_output.cc

```diff
@@ -1069,6 +1069,34 @@ void ProcessFullUse(OneUse* use,
   if (use->ignore_use())   // we're already ignoring it
     return;
 
+  // We normally ignore uses for builtins, but when there is a mapping defined
+  // for the symbol, we should respect that.  So, we need to determine whether
+  // the symbol has any mappings.
+  bool is_builtin_function = false;
+  if (const clang::IdentifierInfo* iden = use->decl()->getIdentifier()) {
+    is_builtin_function = iden->getBuiltinID() != 0 &&
+        !clang::Builtin::Context::isBuiltinFunc(use->symbol_name().c_str());
+  }
+
+  bool is_builtin_function_with_mappings = false;
+
+  if (is_builtin_function) {
+    for (const std::string& public_file :
+        GlobalIncludePicker().
+          GetCandidateHeadersForSymbol(use->symbol_name())) {
+      // To avoid cyclic includes we must only count files that do not
+      // include this one.
+      const FileEntry* public_file_entry =
+        preprocessor_info->IncludeToFileEntry(public_file);
```

Comment:
Continued lines should be indented four spaces from the line above. Clang-format on the changed lines might help here.